src/target/renesas.c

```diff
@@ -78,17 +78,142 @@ typedef enum {
 	PNR_SERIES_RA4M2 = PNR_SERIES('A', '4', 'M', '2'),
 	PNR_SERIES_RA4M3 = PNR_SERIES('A', '4', 'M', '3'),
 	PNR_SERIES_RA4E1 = PNR_SERIES('A', '4', 'E', '1'),
+	PNR_SERIES_RA4E2 = PNR_SERIES('A', '4', 'E', '2'),
 	PNR_SERIES_RA4W1 = PNR_SERIES('A', '4', 'W', '1'),
 	PNR_SERIES_RA6M1 = PNR_SERIES('A', '6', 'M', '1'),
 	PNR_SERIES_RA6M2 = PNR_SERIES('A', '6', 'M', '2'),
 	PNR_SERIES_RA6M3 = PNR_SERIES('A', '6', 'M', '3'),
 	PNR_SERIES_RA6M4 = PNR_SERIES('A', '6', 'M', '4'),
 	PNR_SERIES_RA6M5 = PNR_SERIES('A', '6', 'M', '5'),
 	PNR_SERIES_RA6E1 = PNR_SERIES('A', '6', 'E', '1'),
+	PNR_SERIES_RA6E2 = PNR_SERIES('A', '6', 'E', '2'),
 	PNR_SERIES_RA6T1 = PNR_SERIES('A', '6', 'T', '1'),
 	PNR_SERIES_RA6T2 = PNR_SERIES('A', '6', 'T', '2'),
 } pnr_series_t;
 
+#define NONE 0xffffffff
+
+typedef enum {
+	FLASH_MF3,
+	FLASH_MF4,
+	FLASH_RV40,
+} renesas_ra_flash_type_t;
+
+typedef enum {
+	PNR_FL1,
+	PNR_FL2,
+	PNR_FRT,
+} renesas_ra_pnr_location;
+
+typedef struct renesas_ra_family_details {
+	uint32_t option_start;
+	uint32_t option_size;
+	uint32_t option_start_2;
+	uint32_t option_size_2;
+	uint32_t data_flash_start;
+	uint32_t data_flash_end;
+	uint32_t factory_flash_start;
+	uint32_t factory_size;
+	uint32_t sramhs_start;
+	uint32_t sramhs_end;
+	uint32_t sram0_start;
+	uint32_t sram0_end;
+	uint32_t sram1_start;
+	uint32_t sram1_end;
+	uint32_t stdby_sram_start;
+	uint32_t stdby_sram_end;
```

Comment:
replace `_end` addresses with size, I missed that In the table I shared, we might be able to store only 16b for sizes at least on some variables